rfcs/0000-restructure.md

```diff
@@ -0,0 +1,61 @@
+# Restructure Pipenv Buildpack 
+
+## Proposal
+
+The existing Pipenv buildpack should be rewritten and restructured to *only*
+provide the `pipenv` dependency. Any app dependency installation logic should
+be removed.
+
+## Motivation
+
+In keeping with the overarching [Python Buildpack Rearchitecture
+RFC](https://github.com/paketo-community/python/blob/main/rfcs/0001-restructure.md),
+the Pipenv Buildpack should perform one task, which is installing the `pipenv`
+dependency. This is part of the effort in Paketo Buildpacks to reduce the
+responsibilities of each buildpack to make them easier to understand
+and maintain.
+
+## Implementation
+
+### API
+
+provides: `pipenv`
+requires: `cpython` and `pip` during build
+
+### Detection
+
+The buildpack will pass detection if a Pipfile exists in the app directory. If
+a Pipfile.lock is present, check for a Python version and add it to the
+requirements.
+
+### Configuration
+
+#### BP_PIPENV_VERSION
+
+Users may specify a Pipenv version through the `BP_PIPENV_VERSION` environment
+variable. This can be set explicitly at build time (e.g. `pack --env`) or through
+`project.toml`.
+
+### Dependency Installation
+
+During the build phase, pipenv installation proceeds as outlined below:
+
+Using the pip installation from the preceding `pip` buildpack, download the
+selected Pipenv dependency and extract it into a temporary directory
+(`path/to/pipenv/dependency`).
+
+Run `PYTHONUSERBASE=<path/to/pipenv/layer/> pip install
+<path/to/pipenv/dependency> --user` to install the requested version. Setting
+the `PYTHONUSERBASE` variable ensures that pipenv is installed to the newly
+created layer.
+
+Once the installation is complete, the buildpack:
+
+1. Prepends the pipenv installation (`/layers/<pipenv-layer>/bin`) to `PATH`
+   variable so that it may be invoked without the `python -m` prefix.
+
```

Comment:
I believe the buildpack shouldn't have to prepend to PATH as this should be done for the buildpack by the lifecycle. The buildpack should simply have to place the executable at \<layer\>/bin.

https://github.com/buildpacks/spec/blob/buildpack/v0.6/buildpack.md#layer-paths